crates/block-producer/src/batch_builder/mod.rs

```diff
@@ -1,202 +1,195 @@
 
-        // from the list of unauthenticated notes in the candidate batch, filter out any note
-        // which is also an output note either in any of the ready batches or in the candidate
-        // batch itself
-        txs.iter()
-            .flat_map(|tx| tx.get_unauthenticated_notes().map(|note| note.id()))
-            .filter(|note_id| !all_output_notes.remove(note_id))
-            .collect()
+                Ok((id, batch))
+            }
+        });
     }
-}
 
-#[async_trait]
-impl<S, BB> BatchBuilder for DefaultBatchBuilder<S, BB>
-where
-    S: Store,
-    BB: BlockBuilder,
-{
     #[instrument(target = "miden-block-producer", skip_all, err, fields(batch_id))]
-    async fn build_batch(&self, txs: Vec<ProvenTransaction>) -> Result<(), BuildBatchError> {
+    fn build_batch(
+        txs: Vec<AuthenticatedTransaction>,
+    ) -> Result<TransactionBatch, BuildBatchError> {
         let num_txs = txs.len();
 
         info!(target: COMPONENT, num_txs, "Building a transaction batch");
         debug!(target: COMPONENT, txs = %format_array(txs.iter().map(|tx| tx.id().to_hex())));
 
-        // make sure that all unauthenticated notes in the transactions of the proposed batch
-        // have been either created in any of the ready batches (or the batch itself) or are
-        // already in the store
-        //
-        // TODO: this can be optimized by first computing dangling notes of the batch itself,
-        //       and only then checking against the other ready batches
-        let dangling_notes = self.find_dangling_notes(&txs).await;
-        let found_unauthenticated_notes = match dangling_notes.is_empty() {
-            true => Default::default(),
-            false => {
-                let stored_notes =
-                    match self.store.get_note_authentication_info(dangling_notes.iter()).await {
-                        Ok(stored_notes) => stored_notes,
-                        Err(err) => return Err(BuildBatchError::NotePathsError(err, txs)),
-                    };
-                let missing_notes: Vec<_> = dangling_notes
-                    .into_iter()
-                    .filter(|note_id| !stored_notes.contains_note(note_id))
-                    .collect();
-
-                if !missing_notes.is_empty() {
-                    return Err(BuildBatchError::UnauthenticatedNotesNotFound(missing_notes, txs));
-                }
-
-                stored_notes
-            },
-        };
-
-        let batch = TransactionBatch::new(txs, found_unauthenticated_notes)?;
+        // TODO: This is a deep clone which can be avoided by change batch building to using
+        // refs or arcs.
+        let txs = txs
+            .iter()
+            .map(AuthenticatedTransaction::raw_proven_transaction)
+            .cloned()
+            .collect();
+        // TODO: Found unauthenticated notes are no longer required.. potentially?
+        let batch = TransactionBatch::new(txs, Default::default())?;
```

Comment:
Related to one of the previous comments, I think this may need to be reverted in a future PR - do we have an issue for this?

Reply:
Previous comment [here](https://github.com/0xPolygonMiden/miden-node/pull/562#discussion_r1870441700) and added #566.